Remotion/ObjectBinding/Validation/BusinessObjectValidationResult.cs

```diff
@@ -48,29 +54,78 @@ private static BusinessObjectValidationFailure CreateBusinessObjectValidationFai
       {
         case PropertyValidationFailure propertyValidationFailure
             when validationFailure.ValidatedObject is IBusinessObject validatedBusinessObject
-                 && GetBusinessObjectProperty(validatedBusinessObject, propertyValidationFailure.ValidatedProperty) is IBusinessObjectProperty validatedBusinessObjectProperty:
+                 && GetBusinessObjectProperty(validatedBusinessObject, propertyValidationFailure.ValidatedProperty) is { } validatedBusinessObjectProperty:
         {
-          return BusinessObjectValidationFailure.CreateForBusinessObjectProperty(
-              propertyValidationFailure.LocalizedValidationMessage,
-              validatedBusinessObject,
-              validatedBusinessObjectProperty);
+          return EnumerableUtility.Singleton(
+              BusinessObjectValidationFailure.CreateForBusinessObjectProperty(
+                  propertyValidationFailure.LocalizedValidationMessage,
+                  validatedBusinessObject,
+                  validatedBusinessObjectProperty));
         }
 
         case ObjectValidationFailure objectValidationFailure
-            when validationFailure.ValidatedObject is IBusinessObject validatedBusinessObject:
+            when objectValidationFailure.ValidatedProperties.Count > 0:
         {
-          return BusinessObjectValidationFailure.CreateForBusinessObject(
-              validatedBusinessObject,
+          return CreateBusinessObjectValidationFailuresFromValidatedProperties(
+              objectValidationFailure.ValidatedProperties,
               objectValidationFailure.LocalizedValidationMessage);
         }
 
+        case ObjectValidationFailure objectValidationFailure
+            when validationFailure.ValidatedObject is IBusinessObject validatedBusinessObject:
+        {
+          return EnumerableUtility.Singleton(
+              BusinessObjectValidationFailure.CreateForBusinessObject(
+                  validatedBusinessObject,
+                  objectValidationFailure.LocalizedValidationMessage));
+        }
+
         default:
         {
-          return BusinessObjectValidationFailure.Create(validationFailure.LocalizedValidationMessage);
+          return EnumerableUtility.Singleton(BusinessObjectValidationFailure.Create(validationFailure.LocalizedValidationMessage));
         }
       }
     }
 
+    private static IEnumerable<BusinessObjectValidationFailure> CreateBusinessObjectValidationFailuresFromValidatedProperties (
+        IEnumerable<ValidatedProperty> validatedProperties,
+        string localizedValidationMessage)
+    {
+      var iterated = false;
```

Comment:
naming: should we rename this to hasResult or hasEmptyResult? Something along those lines. 

`iterated` alone is misleading because it could also mean that we have a not-empty list of validatedProperties and not that we've found one or more items to yield return from the method.